include/alpaka/core/Concepts.hpp

```diff
@@ -0,0 +1,53 @@
+/* Copyright 2019 Benjamin Worpitz
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <type_traits>
+
+namespace alpaka
+{
+    namespace concepts
+    {
+        //#############################################################################
+        //! Tag used in class inheritance hierarchies that describes that a specific concept (TConcept)
+        //! is implemented by the given base class (TBase).
+        template<
+            typename TConcept,
+            typename TBase>
+        struct ImplementsConcept
```

Comment:
Just as a thought. Perhaps `ImplementsConcept` is a little wordy, given it is always used like `concepts::ImplementsConcept`. Maybe worth considering to shorten to just `Implements`? Some other languages have `implements` as a keyword for interfaces which in some way are inbetween of C++ interfaces and concepts.

Reply:
I like the idea to shorten the name

Reply:
Done ;-) (I had it this way first)